diff -Naurp u-boot-source.orig/arch/arm/dts/rk3399-rock-pi-4.dts u-boot-source.patched/arch/arm/dts/rk3399-rock-pi-4.dts
--- u-boot-source.orig/arch/arm/dts/rk3399-rock-pi-4.dts	2020-02-29 15:14:00.922456128 +0100
+++ u-boot-source.patched/arch/arm/dts/rk3399-rock-pi-4.dts	2020-03-05 23:02:35.493559963 +0100
@@ -73,7 +73,6 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&vcc5v0_host_en>;
 		regulator-name = "vcc5v0_host";
-		regulator-always-on;
 		vin-supply = <&vcc5v0_sys>;
 	};
 
@@ -84,7 +83,6 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&vcc5v0_typec_en>;
 		regulator-name = "vcc5v0_typec";
-		regulator-always-on;
 		vin-supply = <&vcc5v0_sys>;
 	};
 
@@ -510,10 +508,16 @@
 
 &tcphy0 {
 	status = "okay";
+	tcphy0_usb3: usb3-port {
+                phy-supply = <&vcc5v0_typec>;
+        };
 };
 
 &tcphy1 {
 	status = "okay";
+	tcphy1_usb3: usb3-port {
+                phy-supply = <&vcc5v0_host>;
+        };
 };
 
 &tsadc {
@@ -577,7 +581,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
-	dr_mode = "otg";
+	dr_mode = "host";
 };
 
 &usbdrd3_1 {
