diff -Naurp u-boot-source.orig/arch/arm/dts/rk3399-rock-pi-4-u-boot.dtsi u-boot-source.patched/arch/arm/dts/rk3399-rock-pi-4-u-boot.dtsi
--- u-boot-source.orig/arch/arm/dts/rk3399-rock-pi-4-u-boot.dtsi	2020-02-29 15:14:00.922456128 +0100
+++ u-boot-source.patched/arch/arm/dts/rk3399-rock-pi-4-u-boot.dtsi	2020-03-08 18:39:22.758510023 +0100
@@ -8,7 +8,7 @@
 
 / {
 	chosen {
-		u-boot,spl-boot-order = "same-as-spl", &sdhci, &sdmmc;
+		u-boot,spl-boot-order = "same-as-spl", &sdmmc, &sdhci;
 	};
 };
 
diff -Naurp u-boot-source.orig/include/configs/rockchip-common.h u-boot-source.patched/include/configs/rockchip-common.h
--- u-boot-source.orig/include/configs/rockchip-common.h	2020-02-29 15:14:01.622444242 +0100
+++ u-boot-source.patched/include/configs/rockchip-common.h	2020-03-08 18:34:58.096149409 +0100
@@ -11,11 +11,11 @@
 
 #ifndef CONFIG_SPL_BUILD
 
-/* First try to boot from SD (index 0), then eMMC (index 1) */
+/* First try to boot from SD (index 1), then eMMC (index 0) */
 #if CONFIG_IS_ENABLED(CMD_MMC)
 	#define BOOT_TARGET_MMC(func) \
-		func(MMC, mmc, 0) \
-		func(MMC, mmc, 1)
+		func(MMC, mmc, 1) \
+		func(MMC, mmc, 0)
 #else
 	#define BOOT_TARGET_MMC(func)
 #endif
@@ -39,8 +39,8 @@
 #endif
 
 #define BOOT_TARGET_DEVICES(func) \
-	BOOT_TARGET_MMC(func) \
 	BOOT_TARGET_USB(func) \
+	BOOT_TARGET_MMC(func) \
 	BOOT_TARGET_PXE(func) \
 	BOOT_TARGET_DHCP(func)
 
@@ -51,11 +51,11 @@
 #endif
 #define PARTS_DEFAULT \
 	"uuid_disk=${uuid_gpt_disk};" \
-	"name=loader1,start=32K,size=4000K,uuid=${uuid_gpt_loader1};" \
-	"name=loader2,start=8MB,size=4MB,uuid=${uuid_gpt_loader2};" \
-	"name=trust,size=4M,uuid=${uuid_gpt_atf};" \
-	"name=boot,size=112M,bootable,uuid=${uuid_gpt_boot};" \
-	"name=rootfs,size=-,uuid="ROOT_UUID
+	"name=loader1,start=32K,size=3552K,uuid=${uuid_gpt_loader1};" \
+	"name=ubootenv,start=4064K,size=32K,uuid=${uuid_gpt_ubootenv};" \
+	"name=loader2,start=8MB,size=4MB,uuid=$${uuid_gpt_loader2};" \
+	"name=boot,size=12M,size=128M,bootable,uuid=${uuid_gpt_boot};" \
+	"name=lvm,start=140M,size=-,uuid=${uuid_gpt_lvm}"
 
 #endif
 
