diff -Naurp orig/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi patched/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi
--- orig/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi	2020-12-13 23:41:30.000000000 +0100
+++ patched/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dtsi	2020-12-21 16:26:40.990050056 +0100
@@ -204,7 +204,6 @@
 		clock-output-names = "xin32k", "rk808-clkout2";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pmic_int_l>;
-		rockchip,system-power-controller;
 		wakeup-source;
 
 		vcc1-supply = <&vcc5v0_sys>;
