diff -Naurp atf-source-v2.5/plat/rockchip/rk3399/drivers/dram/dram.h atf-source-v2.5.patched/plat/rockchip/rk3399/drivers/dram/dram.h
--- atf-source-v2.5/plat/rockchip/rk3399/drivers/dram/dram.h	2021-07-26 17:49:44.009118035 +0000
+++ atf-source-v2.5.patched/plat/rockchip/rk3399/drivers/dram/dram.h	2021-07-26 17:50:54.285841798 +0000
@@ -149,7 +149,7 @@ struct rk3399_sdram_params {
 	uint32_t rx_cal_dqs[2][4];
 };
 
-extern __sramdata struct rk3399_sdram_params sdram_config;
+extern struct rk3399_sdram_params sdram_config;
 
 void dram_init(void);
 
