diff -Naurp nor.orig/drivers/mtd/spi-nor/spi-nor.c nor.patched/drivers/mtd/spi-nor/spi-nor.c
--- nor.orig/drivers/mtd/spi-nor/spi-nor.c	2020-03-05 23:12:37.100249964 +0100
+++ nor.patched/drivers/mtd/spi-nor/spi-nor.c	2020-03-07 16:06:18.777695498 +0100
@@ -2503,6 +2503,10 @@ static const struct flash_info spi_nor_i
 	/* XMC (Wuhan Xinxin Semiconductor Manufacturing Corp.) */
 	{ "XM25QH64A", INFO(0x207017, 0, 64 * 1024, 128, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ "XM25QH128A", INFO(0x207018, 0, 64 * 1024, 256, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+
+	/* XTX SPI Nor Flash */
+	{ "xt25f32b", INFO(0x0b4016, 0, 64 * 1024, 64, SECT_4K) },
+	{ "xt25f128b", INFO(0x0b4018, 0, 64 * 1024, 256, SECT_4K) },
 	{ },
 };
 
diff -Naurp dts.orig/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts dts.patched/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts
--- dts.orig/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts	2019-11-25 01:32:01.000000000 +0100
+++ dts.patched/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts	2020-03-07 16:23:41.637308156 +0100
@@ -414,6 +414,27 @@
 	};
 };
 
+&spi1 {
+	spiflash: spi-flash@0 {
+		compatible = "jedec,spi-nor";
+		#address-cells = <0x1>;
+		#size-cells = <1>;
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+		status = "disabled";
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			loader@0 {
+				label = "loader";
+				reg = <0x0 0x400000>;
+			};
+		};
+	};
+};
+			
 &i2c1 {
 	i2c-scl-rising-time-ns = <300>;
 	i2c-scl-falling-time-ns = <15>;
